### ISSUE TITLE
[Feature request] Standardize error handling

### ISSUE THREAD

**Issue by cbourjau** [onnx/onnx#5573]
### System information

NA

### What is the problem that this feature solves?

Certain operators in the ONNX standard have no meaningful way to proceed given some runtime-defined inputs. A simple example of such an operator is `Cast`. Currently, the standard specifies that casting "Hello world" to an integer is undefined / runtime implementation-defined behavior. The onnxruntime 1.15.1 aborts the inference but the standard does not mandate this. Other parts of the standard equally allow for user input to quietly trigger undefined behavior. This introduces uncertainty into the portability and longevity of ONNX models since outputs may differ between runtimes and even versions of the same runtime.

### Describe the feature

A standardized error behavior would eliminate "undefined" behavior providing better interoperability between runtimes and safer updates. Is there an appetite in the community to expand the standard in this direction? If so, I would be willing to work out a proper RFC for this feature.

### Will this influence the current api (Y/N)?

Yes

### Feature Area

Operators

### Are you willing to contribute it (Y/N)

Yes

**Comment by xadupre** [comment]
Should we add failing examples to the backend test series? That way, we would make sure any onnxruntime can be tested against these unsupported cases and continue to fail with the same error message?

**Comment by cbourjau** [comment]
> Should we add failing examples to the backend test series? That way, we would make sure any onnxruntime can be tested against these unsupported cases and continue to fail with the same error message?

Yes, I think providing error test cases would be very useful even if the behavior is not standardized going forward! It might be hard to standardize on an error message, but there should be some way for the framework to specify that a prediction is not covered by the standard / was aborted.
